account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -26,3 +47,149 @@ export function alchemyGasManagerMiddleware(
+      const paymasterData =
+        "0xfffffffffffffffffffffffffffffff0000000000000000000000000000000007aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa1c";
+
+      return {
+        ...uo,
+        dummyPaymasterAndData: {
+          paymaster,
+          paymasterData,
+        },
+      };
+    },
+    feeEstimator: noopMiddleware,
+    gasEstimator: noopMiddleware,
+    paymasterAndData: async (
+      uo,
+      { account, client, feeOptions, overrides: overrides_ }
+    ) => {
+      if (!isAlchemySmartAccountClient(client)) {
+        throw new Error(
+          "AlchemyGasManagerMiddleware can only be used with AlchemySmartAccountClient"
+        );
+      }
+
+      if (!client.chain) {
+        throw new ChainNotFoundError();
+      }
+
+      const userOp = deepHexlify(await resolveProperties(uo));
+
+      const overrides: UserOperationOverrides = filterUndefined({
+        maxFeePerGas: overrideField(
+          "maxFeePerGas",
+          overrides_ as UserOperationOverrides,
+          feeOptions,
+          userOp
+        ),
+        maxPriorityFeePerGas: overrideField(
+          "maxPriorityFeePerGas",
+          overrides_ as UserOperationOverrides,
+          feeOptions,
+          userOp
+        ),
+        callGasLimit: overrideField(
+          "callGasLimit",
+          overrides_ as UserOperationOverrides,
+          feeOptions,
+          userOp
+        ),
+        verificationGasLimit: overrideField(
+          "verificationGasLimit",
+          overrides_ as UserOperationOverrides,
+          feeOptions,
+          userOp
+        ),
+        preVerificationGas: overrideField(
+          "preVerificationGas",
+          overrides_ as UserOperationOverrides,
+          feeOptions,
+          userOp
+        ),
+      });
+
+      const entrypoint = account.getEntryPoint();
+
+      const result = await client.request({
+        method: "alchemy_requestGasAndPaymasterAndData",
+        params: [
+          {
+            policyId,
+            entryPoint: entrypoint.address,
+            userOperation: userOp,
+            dummySignature: await account.getDummySignature(),
+            overrides,
+          },
+        ],
+      });
+
+      return {
+        ...uo,
+        ...result,
+      };
+    },
+  };
+}
+
+/**
+ * Utility function to override a field in the user operation request with the overrides or fee options
+ *
+ * @template {EntryPointVersion} TEntryPointVersion
+ * @param {keyof UserOperationFeeOptions<TEntryPointVersion>} field the field to override
+ * @param {UserOperationOverrides<TEntryPointVersion> | undefined} overrides the overrides object
+ * @param {UserOperationFeeOptions<TEntryPointVersion> | undefined} feeOptions the fee options object from the client
+ * @param {UserOperationRequest<TEntryPointVersion>} userOperation the user operation request
+ * @returns {Hex | Multiplier | undefined} the overridden field value
+ */
+const overrideField = <
```

Comment:
borrowed this from here: https://github.com/alchemyplatform/aa-sdk/blob/v3.19.0/packages/alchemy/src/middleware/gasManager.ts#L257